libs/community/tests/integration_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,838 @@
+        self, aerospike: Aerospike, admin_client: Any
+    ) -> None:
+        """Test max marginal relevance search."""
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz", "bay", "bax", "baw", "bav"],
+            ids=["1", "2", "3", "4", "5", "6", "7"],
+            index_name=index_name,
+            set_name=set_name,
+        )
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=3
+        )
+        sim_output = aerospike.similarity_search("foo", index_name=index_name, k=3)
+
+        assert len(mmr_output) == 3
+        assert mmr_output == sim_output
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=2, fetch_k=3
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "bar"
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo",
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
+
+    def test_cosine_distance(self, aerospike: Aerospike, admin_client: Any) -> None:
+        """Test cosine distance."""
+        from aerospike_vector_search import types
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+            vector_distance_metric=types.VectorDistanceMetric.COSINE,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz"],
+            ids=["1", "2", "3"],
+            index_name=index_name,
+            set_name=set_name,
+        )  # blocking
+
+        """
+        foo vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 0.0]
+        far vector = [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 3.0]
+        cosine similarity ~= 0.71
+        cosine distance ~= 1 - cosine similarity = 0.29
+        """
+        expected = pytest.approx(0.292, abs=0.002)
+        output = aerospike.similarity_search_with_score(
+            "far", index_name=index_name, k=3
+        )
+
+        _, actual_score = output[2]
+
+        assert actual_score == expected
```

Comment:
**Security Issue**: The test asserts exact values for cosine similarity and distance which might not be robust due to floating-point arithmetic precision issues. This could lead to test failures under different environments or changes in underlying libraries. <br> **Fix**: Use approximate comparisons for floating-point numbers to avoid precision issues. <br> **Code Suggestion**: 
 ```
 -        assert actual_score == expected
 +        assert math.isclose(actual_score, expected, rel_tol=1e-9), "Check that actual_score is close to expected with a tolerance"
 ```